### PR TITLE
RTD improvements

### PR DESCRIPTION
**Pull request by henryre**
* Switch to autosummary from apidoc
  * Makes documentation much more legible
  * New page for every member, so more indexable
  * Requires creation of package page source files, handled by sync_api_docs.py
  * This has to be run manually and committed, however it is done by `tox -e fix` and checked by `tox -e check`
  * Currently documents everything importable via `from snorkel.XXX import YYY`, plus additional ones specified in packages.json (see #1358 for reasoning)
* Update Sphinx packages, simplify mocking
* Add logo to RTD, fix some paths, dynamic copyright
* Moves tbX as essential package, removes torchtext (unused)

Here's an initial build, ~~but there are issues with mocked packages~~: https://snorkel.readthedocs.io/en/rtd-nice/. compare to https://snorkel.readthedocs.io/en/redux/

~~Contacting RTD folks now to see if we can get them installed correctly.~~

**Test plan**
* Verify check functionality of sync_api_docs manually
* Verify tox commands manually
* Testing RTD compat as we go

### CODE REVIEW

**Comment by henryre** [comment]
@bhancock8 tbX move and classification errors are the same. tbx is on the import path for Trainer, so it's an essential package (core functionality fails if it isn't installed). Is there any way we can change it so that it isn't always imported with trainer if a different logger is used?

**Comment by henryre** [comment]
Fixed with a Sphinx update!

**Comment by henryre** [comment]
https://snorkel.readthedocs.io/en/rtd-nice/index.html

**Comment by henryre** [comment]
@bhancock8 it’s

File: docs/index.rst [hunk 1]
```diff
@@ -10,21 +10,18 @@ If you're looking for technical details on Snorkel's API,
 you're in the right place.
 
 For more narrative walkthroughs of Snorkel fundamentals or
-example use cases, check out our `homepage <http://snorkel.stanford.edu/>`_
+example use cases, check out our `homepage <http://snorkel.org/>`_
```
Comment (vincentschen) [inline docs/index.rst]: 🎉

File: scripts/sync_api_docs.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    for name in dir(package):
+        if name.startswith("_"):
+            continue
+        obj = getattr(package, name)
+        if isinstance(obj, type) or callable(obj):
+            members.append(name)
+    return members
+
+
+def main(check: bool) -> None:
```
Comment (vincentschen) [inline scripts/sync_api_docs.py]: some quick high-level comments about what this file is executing?

Comment (vincentschen) [reply scripts/sync_api_docs.py]: code is mostly self-explaining; this is just for folks who want to quickly get a sense

**Comment by henryre** [comment]
@vincentschen want to land to get everyone synced, but will address all of these in a follow-up

**Comment by bhancock8** [comment]
@henry the better option for TensorboardX is to just use PyTorch's native tensorboard support (as of 1.1, I think?). Also on the Asana tracker for v0.10 atm.

**Pull request merged**